server/src/database/commands/clean-inactive-workspaces.command.ts

```diff
@@ -0,0 +1,237 @@
+  })
+  parseDryRun(val: string): boolean {
+    return Boolean(val);
+  }
+
+  // We look for public tables which contains workspaceId and updatedAt columns
+  getRelevantTables() {
+    return Object.keys(this.prismaService.client).filter(
+      (name) =>
+        !name.startsWith('_') &&
+        !name.startsWith('$') &&
+        !name.includes('user') &&
+        !name.includes('refreshToken') &&
+        !name.includes('workspace'),
+    );
+  }
+
+  async getTableMaxUpdatedAt(table, workspace) {
+    try {
+      return await this.prismaService.client[table].aggregate({
+        _max: { updatedAt: true },
+        where: { workspaceId: { equals: workspace.id } },
+      });
+    } catch (e) {}
+  }
+
+  updateResult(result, workspace, newUpdatedAt) {
+    if (!result.activityReport[workspace.id]) {
+      result.activityReport[workspace.id] = {
+        displayName: workspace.displayName,
+        maxUpdatedAt: null,
+      };
+    }
+    if (
+      newUpdatedAt &&
+      newUpdatedAt._max.updatedAt &&
+      new Date(result.activityReport[workspace.id].maxUpdatedAt) <
+        new Date(newUpdatedAt._max.updatedAt)
+    ) {
+      result.activityReport[workspace.id].maxUpdatedAt =
+        newUpdatedAt._max.updatedAt;
+    }
+  }
+
+  async detectWorkspacesWithSeedDataOnly(result, workspace) {
+    const companies = await this.prismaService.client.company.findMany({
+      select: { name: true, domainName: true, address: true, employees: true },
+      where: { workspaceId: { equals: workspace.id } },
+    });
+    const people = await this.prismaService.client.person.findMany({
+      select: {
+        firstName: true,
+        lastName: true,
+        city: true,
+        email: true,
+        avatarUrl: true,
+      },
+      where: { workspaceId: { equals: workspace.id } },
+    });
+    const pipelineStages =
+      await this.prismaService.client.pipelineStage.findMany({
+        select: {
+          name: true,
+          color: true,
+          position: true,
+          type: true,
+        },
+        where: { workspaceId: { equals: workspace.id } },
+      });
+    const pipelines = await this.prismaService.client.pipeline.findMany({
+      select: {
+        name: true,
+        icon: true,
+        pipelineProgressableType: true,
+      },
+      where: { workspaceId: { equals: workspace.id } },
+    });
+    if (
+      arraysEqual(people, peopleSeed) &&
+      arraysEqual(companies, companiesSeed) &&
+      arraysEqual(pipelineStages, pipelineStagesSeed) &&
+      arraysEqual(pipelines, [pipelinesSeed])
+    ) {
+      result.sameAsSeedWorkspaces[workspace.id] = {
+        displayName: workspace.displayName,
+      };
+    }
+  }
+
+  async findInactiveWorkspaces(result) {
+    const workspaces = await this.prismaService.client.workspace.findMany();
+    const tables = this.getRelevantTables();
+    for (const workspace of workspaces) {
+      await this.detectWorkspacesWithSeedDataOnly(result, workspace);
```

Comment:
it's a bit weird that detectWorkspacesWithSeedDataOnly is directly updating the result whereas getTableMaxUpdatedAt is not

Reply:
it is done by `updateResult` but i can aggregate those two functions